src/module_1/module_1_meteo_api.py

```diff
@@ -1,7 +1,227 @@
-""" This is a dummy example """
+import requests
+import pandas as pd
+import time
+import os
+import matplotlib.pyplot as plt
 
-def main():
-    raise NotImplementedError
 
-if __name__ == "__main__":
-    main()
+# def main():
+#     raise NotImplementedError
+
+# if __name__ == "__main__":
+#     main()
+
+
+def connect_to_api(url, method="GET", headers=None, params=None, data=None):
+    """
+    Connect to an API using the requests library.
+
+    :param url: URL of the API endpoint.
+    :param method: HTTP method (e.g., 'GET', 'POST', 'PUT', 'DELETE', etc.). Default is 'GET'.
+    :param headers: Dictionary of HTTP headers to send with the request.
+    :param params: Dictionary of URL parameters to append to the URL.
+    :param data: Dictionary of data to send in the body of the request (for POST, PUT, DELETE).
+    :return: Response object from requests.
+    """
+    try:
+        response = requests.request(
+            method, url, headers=headers, params=params, data=data
+        )
+        if response.status_code == 429:
+            # Extract the Retry-After header and sleep for that duration
+            retry_after = int(
+                response.headers.get("Retry-After", 10)
+            )  # Default to 60 seconds if header is missing
+            print(f"Rate limit hit. Retrying after {retry_after} seconds.")
+            time.sleep(retry_after)
+        else:
+            return response
+    except requests.exceptions.HTTPError as errh:
```

Comment:
I don't see the point of excepting any Error that the requests library may raise, specially if you are not doing anything special, you are just printing